File: growing-shop-order/src/main/java/com/example/growingshoporder/domain/OrderLine.java
package com.example.growingshoporder.domain;

import com.example.growingshoporder.dto.OrderRequest;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class OrderLine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //    @ManyToOne(targetEntity = Product.class)
//    @JoinColumn(name="product", nullable = false)
    private Long productId;

    @Column(nullable = false)
    @NotNull
    @Min(1)
    private Integer price;

    @Column(nullable = false)
    @NotNull
    @Min(1)
    private Integer quantity;

    private Boolean isDeleted;

    public OrderLine(Long productId, Integer price, Integer quantity) {
        this.productId = productId;
        this.price = price;
        this.quantity = quantity;
        this.isDeleted = false;
    }

    public Long amounts() {
        return price * quantity.longValue();
    }

    public void update(OrderRequest.OrderLineReq req) {
        this.productId = req.getProductId();
        this.price = req.getPrice();
        this.quantity = req.getQuantity();
    }

    public void delete() {
        this.isDeleted = true;
    }
}

File: growing-shop-domain/src/main/kotlin/com/example/domain/order/OrderLine.kt
package com.example.domain.order

import com.example.domain.common.BaseEntity
import com.example.domain.product.Product
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull

@Entity
class OrderLine(
    @ManyToOne
    @JoinColumn(nullable = false)
    val product: Product,

    @Column(nullable = false)
    @NotNull
    @Min(1)
    val price: Int,

    @Column(nullable = false)
    @NotNull
    @Min(1)
    val quantity: Int,

    isDeleted: Boolean
) : BaseEntity() {
    var isDeleted: Boolean = isDeleted
        protected set

    fun amounts(): Long {
        return price * quantity.toLong()
    }

    fun delete() {
        isDeleted = true
    }
}
